### PR TITLE
[PH17] Implement Inzerva, Master of Insights

### PR DESCRIPTION
**Pull request by PurpleCrowbar**
Implements [Inzerva, Master of Insights](https://scryfall.com/card/ph17/2/inzerva-master-of-insights) of 2017 Heroes of the Realm. Adapts and improves upon the code for fateseal (which could really use some polish). Also resolves a minor text issue with putting cards on the top / bottom of another player's library.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/i/InzervaMasterOfInsights.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                TargetCard targets = new TargetCard(0, cards.size(), Zone.LIBRARY, new FilterCard("cards to PUT on the BOTTOM of " + opponent.getName() + "'s library"));
+                controller.chooseTarget(Outcome.Neutral, cards, targets, source, game);
+                if (!controller.canRespond() || !opponent.canRespond()) {
+                    continue;
+                }
+                controller.putCardsOnBottomOfLibrary(new CardsImpl(targets.getTargets()), game, source, true);
+                cards.removeIf(targets.getTargets()::contains);
+
+                controller.putCardsOnTopOfLibrary(cards, game, source, true);
+                controller.setTopCardRevealed(revealed);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/i/InzervaMasterOfInsights.java:97]: here you use controller, but before you use opponent, one must be wrong

File: Mage/src/main/java/mage/players/PlayerImpl.java [hunk 2]
```diff
@@ -975,8 +975,11 @@ public boolean putCardsOnBottomOfLibrary(Cards cardsToLibrary, Game game, Abilit
                 }
             } else {
                 // user defined order
+                UUID cardOwner = cards.getRandom(game).getOwnerId();
                 TargetCard target = new TargetCard(Zone.ALL,
-                        new FilterCard("card ORDER to put on the BOTTOM of your library (last one chosen will be bottommost)"));
+                        new FilterCard("card ORDER to put on the BOTTOM of " +
+                                (cardOwner == playerId ? "your" : game.getPlayer(cardOwner).getName() + "'s") +
```
Comment (xenohedron) [inline Mage/src/main/java/mage/players/PlayerImpl.java:981]: compare uuid with equals

File: Mage/src/main/java/mage/players/PlayerImpl.java [hunk 3]
```diff
@@ -1078,8 +1081,11 @@ public boolean putCardsOnTopOfLibrary(Cards cardsToLibrary, Game game, Ability s
                 }
             } else {
                 // user defined order
+                UUID cardOwner = cards.getRandom(game).getOwnerId();
                 TargetCard target = new TargetCard(Zone.ALL,
-                        new FilterCard("card ORDER to put on the TOP of your library (last one chosen will be topmost)"));
+                        new FilterCard("card ORDER to put on the TOP of " +
+                                (cardOwner == playerId ? "your" : game.getPlayer(cardOwner).getName() + "'s") +
```
Comment (xenohedron) [inline Mage/src/main/java/mage/players/PlayerImpl.java:1087]: need to fix here too

**Review by xenohedron — APPROVED** [review]

File: Mage/src/main/java/mage/players/PlayerImpl.java (hunk 3 above)
Comment (PurpleCrowbar) [reply Mage/src/main/java/mage/players/PlayerImpl.java]: Of course, silly me

**Pull request merged**